File: elixir/anagram/anagram.exs
defmodule Anagram do
  @doc """
  Returns all candidates that are anagrams of, but not equal to, 'base'.
  """
  @spec match(String.t, [String.t]) :: [String.t]
  def match(base, candidates) do
    base_sig = signature(base)

    candidates
    |> Enum.reject(fn(c) -> String.downcase(c) == base end)
    |> Enum.filter(fn(c) -> signature(c) == base_sig end)
  end

  @spec signature(String.t) :: String.t
  def signature(s) do
    s
    |> String.downcase
    |> String.to_char_list
    |> Enum.sort
    |> List.to_string
  end
end

File: erlang/anagram/anagram.erl
-module(anagram).
-export([find/2]).

-spec find(string(), list(string())) -> list(string()).
find(S, Candidates) ->
    Sig = signature(S),
    [C || C <- Candidates, Sig =:= signature(C), S =/= string:to_lower(C)].

-spec signature(string()) -> string().
signature(S) ->
    lists:sort(string:to_lower(S)).
